ext/sockets/sockets.c

```diff
@@ -872,20 +931,18 @@ PHP_FUNCTION(socket_listen)
 /* {{{ Closes a file descriptor */
 PHP_FUNCTION(socket_close)
 {
-	zval		*arg1;
-	php_socket	*php_sock;
+	zval *arg1;
+	php_socket *php_socket;
 
-	if (zend_parse_parameters(ZEND_NUM_ARGS(), "r", &arg1) == FAILURE) {
+	if (zend_parse_parameters(ZEND_NUM_ARGS(), "O", &arg1, socket_ce) == FAILURE) {
 		RETURN_THROWS();
 	}
 
-	if ((php_sock = (php_socket *)zend_fetch_resource(Z_RES_P(arg1), le_socket_name, le_socket)) == NULL) {
-		RETURN_THROWS();
-	}
+	php_socket = Z_SOCKET_P(arg1);
 
-	if (!Z_ISUNDEF(php_sock->zstream)) {
+	if (!Z_ISUNDEF(php_socket->zstream)) {
```

Comment:
Now that the socket can be closed, don't you need to add checks for closed sockets to the other functions?

Reply:
What about setting `zstream` to `UNDEF`? The other functions do already check this.

Reply:
Yes, we should set it to UNDEF, and probably reset bsd_socket to -1 as well. It doesn't look like all functions consistently check for UNDEF though.

Reply:
In any case, we should test calling all the functions after performing a close, to make sure everything errors sufficiently hard.

Reply:
And an exception should be thrown in this case, right?

Reply:
Yeah. Same as it would be now if you used something after socket_close (right?)

Reply:
Yes, makes sense!